static/app/views/performance/database/noDataMessage.tsx

```diff
@@ -1,106 +1,117 @@
+  isDataAvailable?: boolean;
 }
 
 function DivWrapper(props) {
   return <div {...props} />;
 }
 
-export function NoDataMessage({Wrapper = DivWrapper}: Props) {
+export function NoDataMessage({Wrapper = DivWrapper, isDataAvailable}: Props) {
   const {selection, isReady: pageFilterIsReady} = usePageFilters();
 
   const selectedProjectIds = selection.projects.map(projectId => projectId.toString());
 
-  const {data: projectSpanMetricsCounts, isLoading: areSpanMetricCountsLoading} =
-    useProjectSpanMetricCounts({
-      query: 'span.module:db',
-      statsPeriod: SAMPLE_STATS_PERIOD,
-      enabled: pageFilterIsReady,
+  const {projects: outdatedProjects, isFetching: areOutdatedProjectsFetching} =
+    useOutdatedSDKProjects({
       projectId: selectedProjectIds,
+      enabled: pageFilterIsReady && !isDataAvailable,
     });
 
-  const doesAnySelectedProjectHaveMetrics =
-    sumBy(projectSpanMetricsCounts, 'count()') > 0;
-
-  const {ineligibleProjects, isFetching: areIneligibleProjectsFetching} =
-    useIneligibleProjects({
+  const {projects: denylistedProjects, isFetching: areDenylistProjectsFetching} =
+    useDenylistedProjects({
       projectId: selectedProjectIds,
-      enabled: pageFilterIsReady && !doesAnySelectedProjectHaveMetrics,
     });
 
-  const organization = useOrganization();
-
-  const hasMoreIneligibleProjectsThanVisible =
-    ineligibleProjects.length > MAX_LISTED_PROJECTS;
-
-  if (areSpanMetricCountsLoading || areIneligibleProjectsFetching) {
+  // Hide message while fetching outdated projects or denylisted projects
+  if (areOutdatedProjectsFetching || areDenylistProjectsFetching) {
     return null;
   }
 
-  if (!projectSpanMetricsCounts) {
+  // Hide message if all conditions are satisfied
+  if (
+    isDataAvailable &&
+    !areOutdatedProjectsFetching &&
+    !areDenylistProjectsFetching &&
+    outdatedProjects.length === 0 &&
+    denylistedProjects.length === 0
+  ) {
     return null;
   }
 
-  if (doesAnySelectedProjectHaveMetrics) {
-    return null;
-  }
+  return (
+    <Wrapper>
+      {!isDataAvailable &&
+        tct(
+          'No queries found. Try updating your filters, or learn more about performance monitoring for queries in our [documentation:documentation].',
+          {
+            documentation: (
+              <ExternalLink href="https://docs.sentry.io/product/performance/queries/" />
+            ),
+          }
+        )}{' '}
+      {outdatedProjects.length > 0 &&
+        tct('You may be missing data due to outdated SDKs: [projectList].', {
+          projectList: <ProjectList projects={outdatedProjects} />,
+        })}{' '}
+      {denylistedProjects.length > 0 &&
+        tct(
+          'Some of your projects have been omitted from database metrics extraction. Please contact [support]. Omitted projects: [projectList].',
+          {
+            support: (
+              <ExternalLink href="https://sentry.io/support/">support</ExternalLink>
```

Comment:
h: Is there a better link to contact support? This goes to the Zendesk help center, but that doesn't allow folks to email in, so kind of confusing workflow. We should prob also give support heads up that this exists.

Reply:
Going to ask in Slack!

Reply:
Okay, here's the scoop:

- generally, everywhere in the UI, we direct people to that link. There are lots of examples. The page is good because it's a portal to all kinds of info, and it makes slightly more sense for Self-Hosted folks who might need to read those docs
- I could also run `openHelpSearchModal`, which is what happens when you click "Contact Support" in the sidebar. This is good because it's a more direct link to the Support team, and it opens the Zendesk modal if that's available. This is probably my favourite option here, and Support like it 'cause it appends relevant page context
- some parts of the app just link to email support directly, which I guess also works

Reply:
Ideally we actually show something else for self-hosted to direct users to config directly, but I'm not sure if its worth blocking the PR.

I like the `openHelpSearchModal` option too!

Reply:
Done 👌🏻